src/backend/input/mod.rs

```diff
@@ -336,22 +336,27 @@ impl<B: InputBackend> PointerMotionAbsoluteEvent<B> for UnusedEvent {
 /// Touch events are grouped by slots, usually to identify different
 /// fingers on a multi-touch enabled input device. Events should only
 /// be interpreted in the context of other events on the same slot.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub struct TouchSlot {
-    id: u64,
+    id: Option<u32>,
```

Comment:
I've decided to change the `TouchSlot` here because it already was creating some friction inside my compositor when handling touch events and just created more problems when it came to converting the `Option<TouchSlot>` back to a `i32` to send it to the client.

So since I'd assume nobody is using touch yet (since it doesn't work for clients), changing it now will prevent headaches in the future.